kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java

```diff
@@ -0,0 +1,161 @@
+import com.google.common.primitives.Ints;
+import com.metamx.common.logger.Logger;
+import com.metamx.common.scala.net.curator.Disco;
+import com.metamx.tranquility.config.ConfigHelper;
+import com.metamx.tranquility.config.DataSourceConfig;
+import com.metamx.tranquility.druid.DruidLocation;
+import com.metamx.tranquility.finagle.FinagleRegistry;
+import com.metamx.tranquility.finagle.FinagleRegistryConfig;
+import com.metamx.tranquility.kafka.model.MessageCounters;
+import com.metamx.tranquility.kafka.model.TranquilityKafkaConfig;
+import com.metamx.tranquility.tranquilizer.SimpleTranquilizerAdapter;
+import com.metamx.tranquility.tranquilizer.Tranquilizer;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import scala.Predef;
+import scala.Tuple2;
+import scala.collection.JavaConverters;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Pushes events to Druid through Tranquility using the SimpleTranquilizerAdapter.
+ */
+public class TranquilityEventWriter
+{
+  private static final Logger log = new Logger(TranquilityEventWriter.class);
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static final RetryPolicy RETRY_POLICY = new ExponentialBackoffRetry(1000, 500, 30000);
+
+  private final CuratorFramework curator;
+  private final DataSourceConfig<TranquilityKafkaConfig> dataSourceConfig;
+  private final SimpleTranquilizerAdapter<scala.collection.immutable.Map<String, Object>> simpleTranquilizerAdapter;
+
+  private long rejectedLogCount = 0;
+
+  public TranquilityEventWriter(String topic, DataSourceConfig<TranquilityKafkaConfig> dataSourceConfig)
+  {
+    this.dataSourceConfig = dataSourceConfig;
+
+    final int zkTimeout = Ints.checkedCast(
+        dataSourceConfig.config()
+                        .zookeeperTimeout()
+                        .toStandardDuration()
+                        .getMillis()
+    );
+
+    curator = CuratorFrameworkFactory.builder()
+                                     .connectString(dataSourceConfig.config().zookeeperConnect())
+                                     .connectionTimeoutMs(zkTimeout)
+                                     .retryPolicy(RETRY_POLICY)
+                                     .build();
+
+    final FinagleRegistry finagleRegistry = new FinagleRegistry(
+        FinagleRegistryConfig.builder().build(),
+        new Disco(curator, dataSourceConfig.config())
+    );
+
+    final Tranquilizer<scala.collection.immutable.Map<String, Object>> tranquilizer =
+        ConfigHelper.createTranquilizerWithLocation(
+            dataSourceConfig,
+            finagleRegistry,
+            curator,
+            DruidLocation.create(
+                dataSourceConfig.config().druidIndexingServiceName(),
+                dataSourceConfig.config().useTopicAsDataSource()
```

Comment:
hmm, this useTopicAsDataSource feature is kind of cool, but also strange in that it means the `dataSource` in the spec will be ignored. I think that deserves being called out in the docs